### ISSUE TITLE
:AINewChat throws an error (commit introducing the bug identified)

### ISSUE THREAD

**Issue by jamilraichouni** [madox2/vim-ai#81]
Hi!

First: I love this plugin and appreciate it very much. Thank you so much for this great peace of work!

dec808e78845cb981a7d8d23ac1a1c0eac58d114 introduced a bug:

```
Error executing Lua callback: .../share/nvim/lazy/lazy.nvim/lua/lazy/core/handler/cmd.lua:48: function vim_ai#AINewChatRun[2]..<SNR>48_OpenChatWindow, line 4: Vim:E492: Not an editor command: preset_
stack traceback:
        [C]: in function 'cmd'
        .../share/nvim/lazy/lazy.nvim/lua/lazy/core/handler/cmd.lua:48: in function <.../share/nvim/lazy/lazy.nvim/lua/lazy/core/handler/cmd.lua:16>
Press ENTER or type command to continue
```
I commented the code change location.

With 46e3d5a609fdaa2ddb2af6cc5cb4e754b8b39bfa all is fine.

I have too less knowledge about Vimscript to be able to fix that and come with a PR.

All best from Berlin,
Jamil

**Comment by jamilraichouni** [comment]
I have:

`:version`:

```                                                                                                                                                                                                                                 
NVIM v0.9.5
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```

**Comment by Konfekt** [comment]
I beg your pardon. 

Seems like a Neovim particularity, as in Vimscript `a:0 ?` should be the same as `a:0 != 0` since `a:0` is never less than zero. Well, reverting to `a:0 ?` or `a:0 != 0` then.

Could you read the value of `a:0` (say using `echoerr`)? Does `a:0 != 0` instead of `a:0 > 0` work?

**Comment by Konfekt** [comment]
Does it have anything to do with Lazy? What's the error message without using Lazy?

**Comment by jamilraichouni** [comment]
> Could you read the value of `a:0` (say using `echoerr`)? Does `a:0 != 0` instead of `a:0 > 0` work?

The value seems to be 1:

<img width="1208" alt="image" src="https://github.com/madox2/vim-ai/assets/45113557/93e5df25-d7cb-4f48-be90-676b9680ae16">



> Does a:0 != 0 instead of a:0 > 0 work?

When I replace `a:0 > 0` with `a:0 != 0` the error message is more clear:
`E492: Not an editor command: preset_ `

Also of interest:
`AINewChat` throws that error, `AINewChat below` does not.

**Comment by jamilraichouni** [comment]
Ah, another hint: I use `vim-ai` daily and the issue appears since approximately 3-5 days.

**Comment by jamilraichouni** [comment]
> Does it have anything to do with Lazy? What's the error message without using Lazy?

I'll investigate that later today (latest tomorrow) and come back to you.

**Comment by Konfekt** [comment]
I find this mysterious. I replaced `a:0 ?` with `a:0 > 0 ?` because I found it easier to read as a matter of taste. Surely we can revert it to `a:0 ?`. Still, my curiosity is peaked, and I would not mind hearing what Neovim Lazy deems wrong with it.

**Comment by Konfekt** [comment]
This was a bit of a red hering; I accidently passed `q-args`, allowing for an empty string, instead of `f-args`, which drops a void parameter. Should be fixed with #82

**Comment by jamilraichouni** [comment]
Thank you!
